rationale.md

```diff
@@ -2,6 +2,28 @@
 
 This document attempts to explain why various decision for [W3C WoT(Web of Things) Scripting API](https://w3c.github.io/wot-scripting-api/index.html) were made the way they were.
 
-## 1\. Why Constructor is used instead of Factory?
+## Using factories vs constructors
+As discussed in [issue 3](https://github.com/w3c/wot-scripting-api/issues/3), and as suggested [here](https://github.com/w3c/wot-scripting-api/issues/3#issuecomment-283746764), the guidelines are these:
 
-It would be simpler to expose the WoT object with a constructor. Also, for ExposedThing and ConsumedThing we could provide constructors instead of factories. That would be more aligned with ECMAScript best practices (e.g. an ExposedThing object could be created for testing purposes, shaped locally, then exposed).
+> The root WoT object should not be constructible. It represents the UA's magic ability to discover things, similar to how the Navigator object represents the UA's magic ability to do a bunch of stuff. A namespace might be a good replacement here, if it truly has no state.
+
+> Avoid constructor overloads. A true constructor should be something that directly copies the given essential data into internal fields. If there is a way to infer the essential data from some other data, then that should be a factory. So maybe ThingDescription is the essential data, and if we can infer that from a name or URL, then factory should be used (perhaps static factory, e.g. ExposedThing.fromName()).
+
+> The idea of using a builder pattern (first create an X, then call X.expose() on it to turn into another object, or X-with-UA-magic) is rather unidiomatic in JavaScript. The better way to represent something without UA magic is just a dictionary. Having the same class represent two very distinct things is not great.
+
+
+Resolutions:
+- The WoT API uses a namespace object `wot` in the browser, or provided by `require()` in standalone runtimes.
```

Comment:
I agree and this is what I meant in the text. So we need to make this more clear. What about this:

> The WoT API uses a namespace object `wot` in the browser. In non-browser runtimes, implementations may use either a namespace object `wot`, or an API object provided by the `require()` or `import()` mechanisms.